mesa/visualization/mpl_space_drawing.py

```diff
@@ -323,62 +315,84 @@ def draw_hex_grid(
     s_default = (180 / max(space.width, space.height)) ** 2
     arguments = collect_agent_data(space, agent_portrayal, size=s_default)
 
-    # for hexgrids we have to go from logical coordinates to visual coordinates
-    # this is a bit messy.
-
-    # give all even rows an offset in the x direction
-    # give all rows an offset in the y direction
-
-    # numbers here are based on a distance of 1 between centers of hexes
-    offset = math.sqrt(0.75)
+    # Parameters for hexagon grid
+    size = 1.0
+    x_spacing = np.sqrt(3) * size
+    y_spacing = 1.5 * size
 
     loc = arguments["loc"].astype(float)
-
-    logical = np.mod(loc[:, 1], 2) == 0
-    loc[:, 0][logical] += 0.5
-    loc[:, 1] *= offset
+    # Calculate hexagon centers
+    loc[:, 0] = loc[:, 0] * x_spacing + (loc[:, 1] % 2) * (x_spacing / 2)
```

Comment:
So I think something is not right. Look at the top row in both figures. You'll see that they are differently aligned. This means that the neigborhood of a given cell as shown in the grid is not the same as in the underlying data. You can also check this by placing an agent in the very first cell. 

This is due to this line of code: `loc[:, 1] % 2`. Note how I check for 0, rather than 1.

Reply:
This issue is still not resolved.